crates/l2/prover/proof_data_client.rs

```diff
@@ -31,63 +31,73 @@ impl ProofDataClient {
         let prover = SP1Prover::new();
 
         loop {
-            let id = self.request_new_data().unwrap();
-
-            let proof = prover.prove(id).unwrap();
-
-            self.submit_proof(id, proof).unwrap();
-
-            sleep(Duration::from_secs(5)).await;
+            match self.request_new_data() {
```

Comment:
Shouldn't this loop sleep always after trying to request the data?

I mean:
- In the error branch it will immediately try again instead of waiting.
- In the successful branch, it will try again immediately after submitting the proof, also: if the submission of the proof fails, what happens? You've already requested the new data, but does the provider keep it for you in this case?

Reply:
I think in the absolute successful branch it should request immediately, as it has nothing else to do. In case of any error, it could be a good idea to wait, in case it's a network error.
As of the retry on submitting error, it's a TODO right now. Maybe the best thing to do in that case is retry asynchronously, to avoid wasting proving time, although we are not optimizing at that level yet